src/ZkFold/Symbolic/Cardano/Contracts/BatchTransfer.hs

```diff
@@ -26,26 +29,19 @@ type TxOut context = Output Tokens () context
 type TxIn context  = Input Tokens () context
 type Tx context = Transaction 6 0 11 Tokens 0 () context
 
-hash :: forall context x . (Symbolic context, MiMCHash (BaseField context) context x) => x -> FieldElement context
-hash = mimcHash @(BaseField context) mimcConstants zero
-
-type Sig context =
+hash :: forall context x .
     ( Symbolic context
-    , StrictConv (context Par1) (UInt 256 Auto context)
-    , FromConstant Natural (UInt 256 Auto context)
-    , MultiplicativeSemigroup (UInt 256 Auto context)
-    , MiMCHash (BaseField context) context (TxOut context, TxOut context)
-    , Eq (Bool context) (UInt 256 Auto context)
-    , Eq (Bool context) (TxOut context)
-    , Iso (UInt 256 Auto context) (ByteString 256 context)
-    , Extend (ByteString 224 context) (ByteString 256 context))
+    , SymbolicData x
+    , Context x ~ context
+    , Support x ~ Proxy context
+    ) => x -> FieldElement context
+hash = mimcHash @(BaseField context) mimcConstants zero
 
 verifySignature ::
-    forall context . Sig context =>
-    ByteString 224 context ->
-    (TxOut context, TxOut context) ->
-    ByteString 256 context ->
-    Bool context
+    forall context .
+    ( Symbolic context
+    , SymbolicData (TxOut context, TxOut context)
```

Comment:
Can we derive it from `SymbolicData (TxOut context)`?

Reply:
This also needs `KnownNat (TxOut context)` until we refactor `SymbolicData`